packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1470,17 +1525,28 @@ class _RenderDecoration extends RenderBox {
 
     if (label != null) {
       final double labelX = _boxParentData(label!).offset.dx;
+      // +1 shifts the range of x from (-1.0, 1.0) to (0.0, 2.0).
+      final double floatAlign = decoration.floatingLabelAlignment._x + 1;
```

Comment:
Thanks for making the calculations work with any `_x` and not just center and left 👍